modules/dmrpp_module/build_dmrpp_util.cc

```diff
@@ -0,0 +1,963 @@
+
+    // How the fill value is defined?
+    H5D_fill_value_t status;
+    if ( (H5Pfill_value_defined(plist_id, &status)) < 0 ) {
+        H5Pclose(plist_id);
+        throw BESInternalError("Unable to access HDF5 Fillvalue information.", __FILE__, __LINE__);
+    }
+
+    H5Pclose(plist_id);
+
+    return status != H5D_FILL_VALUE_UNDEFINED;
+}
+
+#if 0
+enum H5DataType
+{
+    H5CHAR,
+    H5UCHAR,
+
+    H5INT16,
+    H5UINT16,
+    H5INT32,
+    H5UINT32,
+    H5INT64,
+    H5UINT64,
+
+    H5FLOAT32,
+    H5FLOAT64,
+
+    H5FSTRING,
+    H5VSTRING,
+
+    H5REFERENCE,
+    H5COMPOUND,
+    H5ARRAY,
+    H5UNSUPTYPE
+};
+#endif
+
+/**
+ * @brief Get the HDF5 value as a string
+ *
+ * We need it as a string for the DMR++ XML. Used to add Fill Value
+ * HDF5 attribute information to the chunk elements of the DMR++
+ *
+ * @param h5_type_id
+ * @param value
+ * @return The string representation of the value.
+ */
+string
+get_value_as_string(hid_t h5_type_id, vector<char> &value)
+{
+    ostringstream oss;
+    H5T_class_t class_type = H5Tget_class(h5_type_id);
+    int sign;
+    switch (class_type) {
+        case H5T_INTEGER:
+            sign = H5Tget_sign(h5_type_id);
+            switch (H5Tget_size(h5_type_id)) {
+                case 1:
+                    if (sign == H5T_SGN_2)
+                        oss << *(int8_t *)(value.data());
+                    else
+                        oss << *(uint8_t *)(value.data());
+                    break;
+                case 2:
+                    if (sign == H5T_SGN_2)
+                        oss << *(int16_t *)(value.data());
+                    else
+                        oss << *(uint16_t *)(value.data());
+                    break;
+                case 4:
+                    if (sign == H5T_SGN_2)
+                        oss << *(int32_t *)(value.data());
+                    else
+                        oss << *(uint32_t *)(value.data());
+                    break;
+                case 8:
+                    if (sign == H5T_SGN_2)
+                        oss << *(int64_t *)(value.data());
+                    else
+                        oss << *(uint64_t *)(value.data());
+                    break;
+                default:
+                    throw BESInternalError("Unable extract integer fill value.", __FILE__, __LINE__);
+            }
+            break;
+
+        case H5T_FLOAT:
+            switch (H5Tget_size(h5_type_id)) {
+                case 4:
+                    oss << *(float*)(value.data());
+                    break;
+                case 8:
+                    oss << *(double*)(value.data());
+                    break;
+                default:
+                    throw BESInternalError("Unable extract float fill value.", __FILE__, __LINE__);
+            }
+            break;
+
+        case H5T_STRING:
+            return "unknown"; // FIXME Add support for String fill values. jhrg 4/22/22
+
+        case H5T_ARRAY:
```

Comment:
This is even remote. Array datatype is seldom used and also not the fillvalue.

Reply:
OK, but we have to handle this because some of our test data does have the fill value attribute set for H5T_ARRAY. It's odd, and I have no idea how to get the value. 

What should I do for this case? clear the fill value so that it does not appear to be set?

Reply:
Are you sure? Where do you get the data? 

> OK, but we have to handle this because some of our test data does have the fill value attribute set for H5T_ARRAY. It's odd, and I have no idea how to get the value.

Reply:
You're correct - there are no cases of ARRAY or COMPOUND in the dmrpp builder test data. There are some instances of STRING.